baseplate/server/__init__.py

```diff
@@ -197,3 +202,69 @@ def load_and_run_script():
     config = read_config(args.config_file, server_name=None, app_name=args.app_name)
     configure_logging(config, args.debug)
     args.entrypoint(config.app)
+
+
+def load_and_run_tshell():
+    """Launch a shell for a thrift service."""
+    parser = argparse.ArgumentParser(
+        description="Open a shell with app configuration loaded.",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument("--debug", action="store_true", default=False,
+        help="enable extra-verbose debug logging")
+    parser.add_argument("--app-name", default="main", metavar="NAME",
+        help="name of app to load from config_file (default: main)")
+    parser.add_argument("config_file", type=argparse.FileType("r"),
+        help="path to a configuration file")
+
+    args = parser.parse_args(sys.argv[1:])
+    config = read_config(args.config_file, server_name=None, app_name=args.app_name)
+    configure_logging(config, args.debug)
+
+    env = dict()
+    env_banner = {
+        'app': "This project's app instance",
+        'context': "The context for this shell instance's span",
+    }
+
+    app = make_app(config.app)
+    env['app'] = app
+
+    baseplate = app._event_handler.baseplate
+    span = baseplate.make_server_span(RequestContext(), 'shell')
+    env['context'] = span.context
+
+    if 'setup' in config.tshell:
+        setup = _load_factory(config.tshell['setup'])
+        setup(env, env_banner)
+
+    # generate banner text
+    banner = "Available Objects:\n"
+    for var in sorted(env_banner.keys()):
+        banner += '\n  %-12s %s' % (var, env_banner[var])
+
+    try:
+        # try to use IPython if possible
+        try:
+            try:
+                # 1.0 <= ipython
```

Comment:
Would you be OK dropping the old IPython support? 

> IPython 1.0 was released on August 8, 2013 (http://ipython.org/news.html)

In general we should probably be writing the minimal version of what we need here rather than just copying pylons; they have many more cases (and much longer history) to deal with.

Reply:
Sure thing. Yeah, this was copied from r2 which was copied from pylons.